doc/design/refactor_discuess/net_design.md

```diff
@@ -0,0 +1,65 @@
+# Network Design
```

Comment:
I'd suggest that we move this design doc into `paddle/framework/net.md`, so we could keep the code (implementation) together with the design.